homeassistant/components/netgear/__init__.py

```diff
@@ -52,14 +55,33 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         configuration_url=f"http://{entry.data[CONF_HOST]}/",
     )
 
+    async def async_update_data():
+        """Fetch data from the router."""
+        await router.async_update_device_trackers()
```

Comment:
Then the coordinator will catch the error and set all devices to unavailable, they will become available again on the next update.
(I think)

Reply:
Actually my previous statement is not true (I think).
You would need to catch the error and raise UpdateFailed.
However pynetgear does not raise any known errors and catches all errors (which are known) and logs them and then just returns False or None for the function.
So there is nothing to catch.
Besides setting device_trackers to unavailable is not desired (I think), so we do not need to raise UpdateFailed when the method returns False or None.

Reply:
No, you're right, that function never returns anything (`->None`).

Perhaps this is a bigger question than my initial ask, but what would happen if the update fails for 100 hours and the user relies on the data being updated?

Reply:
@ggravlingen well for one they would have 100*60*2 = 12000 errors logged in their log, so hopefully they will notice that and fix the issue/report the bug.
The state of the devices will keep the same over that 100 hours.
Note that that is the same behaviour as the current integration.
However I am fine with implementing a unavailable state, because failed updates schould just not happen and schould be fixed.

Reply:
I have no opinions on how to handle that, was just curious. It's for the codeowner to decide, I just added a few thoughts to this PR based on what I saw.

Reply:
I am the codeowner, but I am very open to inputs.

Reply:
Let's mark this thread on the update method as resolved. I wasn't able to do that now.